### PR TITLE
tgui: Secrets Panel

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Tgui-izes the secrets panel
### small secret panel changes
Removes one terrible shuttle button, but just know all of them are pretty shit
There can only be one (40 second delay) was condensed into an option for there can only be one

secrets

## Why It's Good For The Game

Before it was a blank HTML document. It has some kind of useful stuff, and also some shitpost-y stuff. Using some tab system really helps you pick out where the logging/issue fixing buttons are and where the silly garbage

## Changelog
:cl:
admin: Secrets panel tgui-ized
/:cl:

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
Admin tools that don't use 8+ year old HTML menus are against god and nature and lack soul

**Comment by tralezab** [comment]
Way more attention than secrets deserves but it's kind of a relic of history so woop woop. Still, planning on trashing the awful debug tools, moving out some stuff that shouldn't be there into general verbs/other stuff, moving in some adminbuse that should be in there, and moving logging into investigate.

But for now a UI update should help people find what they need.

**Comment by optimumtact** [comment]
upvoted for gratuitous use of gauge

File: tgui/packages/tgui/interfaces/Secrets.js [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+      <Flex
+        mb={1}
+        grow={1}
+        direction="row"
+        height="100%"
+        align="stretch"
+        justify="space-between">
+        <Flex.Item>
+          <Button
+            align="center"
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Secrets.js:304]: Align does nothing on Button, if you want to align all text to center, just put textAlign on the most upper level element, and all children will inherit it.

File: tgui/packages/tgui/interfaces/Secrets.js [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+      </Flex.Item>
+    </Flex>
+  );
+};
+
+const HelpfulTab = (props, context) => {
+  const { act } = useBackend(context);
+  return (
+    <Flex direction="column" mb={-0.75} mx={-0.5}>
+      <Flex
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Secrets.js:61]: Have you tried just using wrapping properties of Flex? You would not need to manually split buttons in separate lines, and you would not need to use `direction="column"` as well, just one big flex.

File: tgui/packages/tgui/interfaces/Secrets.js [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+                    onClick={() => act("show_admins")} />
+                </Fragment>)}>
+              <Flex
+                mx={-0.5}
+                align="stretch"
+                justify="center">
+                <Flex.Item bold>
+                  <NoticeBox color="black">
+                    {"The first rule of adminbuse is:\
+                    you don't talk about the adminbuse."}
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Secrets.js:644]: Do not use `{""}`, place text into the tag as is.

File: tgui/packages/tgui/interfaces/Secrets.js [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+  );
+};
+
+export const Secrets = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    is_debugger,
+    is_funmin,
+  } = data;
+  const TAB2NAME = [
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Secrets.js:583]: Extract it to above Secrets function, to not reconstruct this array every time Secrets re-render.

File: tgui/packages/tgui/interfaces/Secrets.js [hunk 5]
```diff
@@ -0,0 +1,723 @@
+import { toFixed } from 'common/math';
+import { Fragment } from 'inferno';
+import { useBackend, useLocalState } from '../backend';
+import { Button, Flex, LabeledControls, NoticeBox, RoundGauge, Section } from '../components';
+import { FlexItem } from '../components/Flex';
+import { Window } from '../layouts';
+
+const formatPressure = value => {
+  return toFixed(value) + '%';
+};
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Secrets.js:10]: Inline this function to where the round gauge is used, i.e.:
```
format={value => toFixed(value) + '%'}
```

**Review by stylemistake — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/Secrets.js (hunk 3 above)
Comment (tralezab) [reply tgui/packages/tgui/interfaces/Secrets.js:644]: Quotes and apostrophes throw lint errors when outside of ellipses,  which is why i did it. Would you know anything about that?

Comment (stylemistake) [reply tgui/packages/tgui/interfaces/Secrets.js:644]: Right... You can use `&apos;`.

**Comment by Dorsisdwarf** [comment]
Looks very cool, hopefully will be more performant than the cool but jankety drop pod ui

File: tgui/packages/tgui/interfaces/Secrets.js (hunk 3 above)
Comment (tralezab) [reply tgui/packages/tgui/interfaces/Secrets.js:644]: Ooooooooh

**Review by stylemistake — DISMISSED** [review]

**Comment by stylemistake** [comment]
Needs a rebuild, but LGTM.

**Pull request merged**